<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <script>
        /* 
            throw
            1.抛出一个异常值(错误)
            2.中断程序执行
            3.约定throw必须抛出一个new Error()对象
            4.new Error(参数)对象，这个参数必须是一个字符串，参数设置给对象中的message属性
            5.throw 抛出的错误，被try catch捕获，将错误的影响范围控制在局部，
              throw抛出的错误将会赋值给catch(e)变量
        */
    //    throw new Error('this is error 100');
    //    console.dir(Error);
       try {

            // console.log(username);
            throw new Error('this is error 100');
       } catch (e) {
            console.log(e.message);
       }

       function a(params) {
        return 0
       }
       console.log(typeof a());

       console.log(new Number(2));
       console.log(2 == new Number(2));
       
    </script>
</body>
</html>